Lib/fontTools/pens/basePen.py

```diff
@@ -36,26 +36,50 @@
 sequence of length 2 will do.
 """
 
-from fontTools.misc.py23 import *
+import abc
+from typing import Any, Tuple
+
 from fontTools.misc.loggingTools import LogMixin
 
 __all__ =  ["AbstractPen", "NullPen", "BasePen",
 			"decomposeSuperBezierSegment", "decomposeQuadraticSegment"]
 
 
-class AbstractPen(object):
-
-	def moveTo(self, pt):
+class AbstractPen(abc.ABC):
+	@classmethod
+	def __subclasshook__(cls, subclass: Any) -> bool:
+		return (
```

Comment:
in the `__subclasshook__` you also need to add a check like `if cls is not AbstractPen: return NotImplemented`, to ensure concrete subclasses of AbstractPen will use the regular `issubclass` logic instead of the custom checks.
You don't want that issubclass return True even with concrete pens.

See
https://stackoverflow.com/questions/35858285/why-check-if-cls-is-the-class-in-subclasshook

Cf collections.abc or the example in the official docs does:
https://docs.python.org/3/library/abc.html#abc.ABCMeta.__subclasshook__